### ISSUE TITLE
Add tag parameter to xml-fold findEnclosingTag

### ISSUE THREAD

**Issue by mce2** [codemirror/CodeMirror#4669]
It would be nice if _findEnclosingTag_ function is able to find enclosing tag of particular type.
Supposed solution is straitforward:
```
  CodeMirror.findEnclosingTag = function(cm, pos, range, tag) {
    var iter = new Iter(cm, pos.line, pos.ch, range);
    for (;;) {
      var open = findMatchingOpen(iter, tag);
```

**Comment by marijnh** [comment]
Could you submit this as a pull request?

**Comment by mce2** [comment]
Sorry, I'm new to git and GitHub. Here it is: https://github.com/codemirror/CodeMirror/pull/4670

**Comment by marijnh** [comment]
Great, merged #4670
